src/group.jl

```diff
@@ -0,0 +1,111 @@
+@mpi_handle Group
+
+const GROUP_NULL = _Group(MPI_GROUP_NULL)
+const GROUP_EMPTY = _Group(MPI_GROUP_EMPTY)
+
+Group() = Group(GROUP_NULL.val)
+
+function free(group::Group)
+    if group.val != GROUP_NULL.val
+        @mpichk ccall((:MPI_Comm_free, libmpi), Cint, (Ptr{MPI_Group},), group)
```

Comment:
I've started pasting the C signatures (from the MPI spec) above the ccalls: I've found it useful to check we're passing arguments correctly.